media-libs/smpeg2/smpeg2-2.0.0.recipe

```diff
@@ -0,0 +1,88 @@
+SUMMARY="An MPEG decoding library"
+DESCRIPTION="SMPEG is short for the SDL MPEG library originally developed \
+by Loki Software. SMPEG uses MPEG-1 standard as a video display library, \
+rather than MPEG-2, since only the MPEG-1 standard is unencumbered with \
+software patents in the United States."
+HOMEPAGE="http://www.icculus.org/smpeg"
+COPYRIGHT="1999-2004 Sam Lantinga, Joe Tennies."
+LICENSE="GNU LGPL v2"
+REVISION="1"
+SOURCE_URI="https://www.libsdl.org/projects/smpeg/release/smpeg2-$portVersion.tar.gz"
+CHECKSUM_SHA256="979a65b211744a44fa641a9b6e4d64e64a12ff703ae776bafe3c4c4cd85494b3"
+
+ARCHITECTURES="x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	smpeg2$secondaryArchSuffix = 2.0.0 compat >= 2
+	cmd:smpeg2 = 2.0.0 compat >= 2
+	cmd:plaympeg$secondaryArchSuffix = 2.0.0 compat >= 2
+	lib:libsmpeg2$secondaryArchSuffix = 2.0_0.0.0 compat >= 2.0_0
```

Comment:
Using the _S option for a while already, no problems so farVerzonden vanaf mijn Huawei mobiele telefoon-------- Oorspronkelijk bericht --------Onderwerp: Re: [haikuports/haikuports] smpeg2 new recipe (#668)Van: fbrosson Aan: haikuports/haikuports Cc: Begasus ,Author In media-libs/smpeg2/smpeg2-2.0.0.recipe:
> +software patents in the United States."
> +HOMEPAGE="http://www.icculus.org/smpeg"
> +COPYRIGHT="1999-2004 Sam Lantinga, Joe Tennies."
> +LICENSE="GNU LGPL v2"
> +REVISION="1"
> +SOURCE_URI="https://www.libsdl.org/projects/smpeg/release/smpeg2-$portVersion.tar.gz"
> +CHECKSUM_SHA256="979a65b211744a44fa641a9b6e4d64e64a12ff703ae776bafe3c4c4cd85494b3"
> +
> +ARCHITECTURES="x86_gcc2 x86 x86_64"
> +SECONDARY_ARCHITECTURES="x86"
> +
> +PROVIDES="
> +	smpeg2$secondaryArchSuffix = 2.0.0 compat >= 2
> +	cmd:smpeg2 = 2.0.0 compat >= 2
> +	cmd:plaympeg$secondaryArchSuffix = 2.0.0 compat >= 2
> +	lib:libsmpeg2$secondaryArchSuffix = 2.0_0.0.0 compat >= 2.0_0

I think it should be just lib:libsmpeg2_2.0$secondaryArchSuffix without any lib versioning because a "2.0" versioning is already in the name of the lib before the ".so". BTW, you might wish to replace the _2.0 by _${portVersion%\.*} to obtain: lib:libsmpeg2_${portVersion%\.*}$secondaryArchSuffix but I don't know if there is a wide acceptance for that. (I've used it in a couple of recipes and it was accepted ;-)

haikuporter has an excellent option that I always use: -S. It enables a strict mode which reports many errors, like, for example, when you forget items for the PROVIDES* or REQUIRES*. So if you forget a dependency on an external lib it will spot it. And if a libname is wrong in the provides, it will also tell you that you don't have a provides for it. I really recommend always using the strict mode. BTW, if you use it on a recipe that has any of the errors it can spot then it will not create the hpkgs, and I like this behaviour ;-)

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.